radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineStateComputer.java

```diff
@@ -125,11 +125,22 @@ public HashCode hash() {
 
 	private BFTValidatorSet executeSystemUpdate(
 		RadixEngineBranch<LedgerAtom> branch,
+		long epoch,
 		View view,
 		long timestamp,
 		ImmutableList.Builder<PreparedCommand> successBuilder
 	) {
 		final SystemParticle lastSystemParticle = branch.getComputedState(SystemParticle.class);
+		if (lastSystemParticle.getEpoch() != epoch) {
```

Comment:
Probably it might have sense to move this `if` into dedicated method with descriptive name, for example `validateSynchronyPresence`.